python/ray/internal/storage.py

```diff
@@ -152,6 +154,28 @@ def delete(self, path: str) -> bool:
         except FileNotFoundError:
             return False
 
+    def delete_prefix(self, path: str) -> bool:
```

Comment:
We should rename it to delete_dir since this won't cover `delete /abc/def/prefix*`